### PR TITLE
GUI: fix model overlay header sync

### PR DESCRIPTION
**Pull request by jonasschnelli**
Update the block and header tip is constraint to have a minimal distance of 250ms between updates... which can lead to miss the last header update.

The modal overlay then assumes we are still in header sync and the view get stuck in "syncing headers,..." (while it's actually syncing blocks).

This removes the 250ms minimal delta for header updates as well as it fixes the correct display of how header updates should update the labels.

### CODE REVIEW

File: src/qt/modaloverlay.cpp [hunk 1]
```diff
@@ -136,11 +137,16 @@ void ModalOverlay::tipUpdate(int count, const QDateTime& blockDate, double nVeri
...
     } else {
-        ui->numberOfBlocksLeft->setText(tr("Unknown. Syncing Headers (%1)...").arg(bestHeaderHeight));
+        UpdateHeaderSyncLabel();
         ui->expectedTimeLeft->setText(tr("Unknown..."));
     }
 }
 
+void ModalOverlay::UpdateHeaderSyncLabel() {
+    int est_headers_left = bestHeaderDate.secsTo(QDateTime::currentDateTime()) / Params().GetConsensus().nPowTargetSpacing;
+    ui->numberOfBlocksLeft->setText(tr("Unknown. Syncing Headers (%1, %2%)...").arg(bestHeaderHeight).arg(QString::number(100.0 / (bestHeaderHeight+est_headers_left)*bestHeaderHeight, 'f', 1)));
```
Comment (hebasto) [inline src/qt/modaloverlay.cpp]: spaces around arithmetic operators?

File: src/qt/clientmodel.cpp [hunk 2]
```diff
@@ -238,7 +238,7 @@ static void BlockTipChanged(ClientModel *clientmodel, bool initialSync, int heig
         clientmodel->cachedBestHeaderTime = blockTime;
     }
     // if we are in-sync, update the UI regardless of last update time
```
Comment (MarcoFalke) [inline src/qt/clientmodel.cpp]: Comment needs update?

**Comment by jonasschnelli** [comment]
> The delta is there to prevent a flood of GUI update messages. I don't removing it is a good solution, we should take care that one update makes it per 250ms though.

I agree. This PR only removes the delay-update for header updates... which IMO should not result in GUI flood. AFAIK the GUI flood only gets problematic on chain tip updates.

File: src/qt/clientmodel.cpp (hunk 2 above)
Comment (jonasschnelli) [reply src/qt/clientmodel.cpp]: Fixed...

**Comment by hebasto** [comment]
tACK e8db6b8044424a6fbb4b0a80acffc8c7fbf6be8b (Linux Mint 19.2).

File: src/qt/modaloverlay.cpp [hunk 3]
```diff
@@ -136,11 +137,16 @@ void ModalOverlay::tipUpdate(int count, const QDateTime& blockDate, double nVeri
...
     } else {
-        ui->numberOfBlocksLeft->setText(tr("Unknown. Syncing Headers (%1)...").arg(bestHeaderHeight));
+        UpdateHeaderSyncLabel();
         ui->expectedTimeLeft->setText(tr("Unknown..."));
     }
 }
 
+void ModalOverlay::UpdateHeaderSyncLabel() {
+    int est_headers_left = bestHeaderDate.secsTo(QDateTime::currentDateTime()) / Params().GetConsensus().nPowTargetSpacing;
+    ui->numberOfBlocksLeft->setText(tr("Unknown. Syncing Headers (%1, %2%)...").arg(bestHeaderHeight).arg(QString::number(100.0 / (bestHeaderHeight + est_headers_left) * bestHeaderHeight, 'f', 1)));
```
Comment (promag) [inline src/qt/modaloverlay.cpp]: nit, could use overloaded `QString::arg` http://doc.qt.io/qt-5/qstring.html#arg-8 instead of `QString::number`.

**Comment by promag** [comment]
utACK e8db6b8.

**Pull request merged**